circus-train-avro/src/test/java/com/hotels/bdp/circustrain/avro/transformation/AbstractAvroSerDeTransformationTest.java

```diff
@@ -0,0 +1,154 @@
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("default"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testOneReplicationsOverrideFailureLifecycle() throws Exception {
+    EventTableReplication tableReplication = mockTableReplication("overrideBaseUrl");
+    runLifeCycleFailure(tableReplication);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("overrideBaseUrl"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsOverrideFailureLifecycle() throws Exception {
+    EventTableReplication tableReplication = mockTableReplication("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat("overrideBaseUrl", is(transformation.getAvroSchemaDestinationFolder()));
+    assertThat(EVENT_ID, is(transformation.getEventId()));
+    assertThat("location", is(transformation.getTableLocation()));
+    EventTableReplication tableReplication2 = mockTableReplication("overrideBaseUrl2");
+    runLifeCycleFailure(tableReplication2);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("overrideBaseUrl2"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsSecondOverrideShouldUseDefaultFailureLifecycle() throws Exception {
+    EventTableReplication tableReplication = mockTableReplication("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat("overrideBaseUrl", is(transformation.getAvroSchemaDestinationFolder()));
+    assertThat(EVENT_ID, is(transformation.getEventId()));
+    assertThat("location", is(transformation.getTableLocation()));
+    EventTableReplication tableReplication2 = mockTableReplication(null);
+    runLifeCycleFailure(tableReplication2);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("default"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  private EventTableReplication mockTableReplication(String overrideBaseUrl) {
+    EventTableReplication result = Mockito.mock(EventTableReplication.class);
+    Map<String, Object> transformOptions = new HashMap<>();
+    if (overrideBaseUrl != null) {
+      Map<String, Object> avroOverrideOptions = new HashMap<>();
+      avroOverrideOptions.put(AvroSerDeConfig.TABLE_REPLICATION_OVERRIDE_BASE_URL, overrideBaseUrl);
+      transformOptions.put(AvroSerDeConfig.TABLE_REPLICATION_OVERRIDE_AVRO_SERDE_OPTIONS, avroOverrideOptions);
+    }
+    when(result.getTransformOptions()).thenReturn(transformOptions);
+    EventReplicaTable eventReplicaTable = new EventReplicaTable("db", "table", "location");
```

Comment:
Minor, but you could consider making "location" a constant since it's used all over the show, when reading from top to bottom I didn't know where the value was coming from until I got here.

Reply:
yes will do